lib/event_store/storage/appender.ex

```diff
@@ -117,10 +108,24 @@ defmodule EventStore.Storage.Appender do
     event_id |> uuid()
   end
 
-  defp insert_event_batch(conn, events, schema, opts) do
-    event_count = length(events)
-    statement = Statements.insert_events(schema, event_count)
-    parameters = build_insert_parameters(events)
+  defp insert_event_batch(
+         conn,
+         stream_id,
+         stream_uuid,
+         events,
+         event_count,
+         schema,
+         any_version,
+         opts
+       ) do
+    statement =
+      case any_version do
+        true -> Statements.insert_events_any_version(schema, stream_id, event_count)
+        false -> Statements.insert_events(schema, stream_id, event_count)
+      end
+
+    stream_id_or_uuid = stream_id || stream_uuid
+    parameters = [stream_id_or_uuid | [event_count | build_insert_parameters(events)]]
```

Comment:
`[stream_id_or_uuid, event_count] ++ build_insert_parameters(events)` is probably equivalent in terms of performance and a bit more legible, IMO

Reply:
It could be benchmarked to see whether there's difference. The length of the insert parameters list is seven times the number of events, so a micro optimisation was used here over readability.

Reply:
It will make no difference, since lists are singly linked, only `a` from `a ++ b` has to be relinked. `b` remains the same. I did benchmark it just now to be sure, doing `[1, 2 | 500_items]` takes 2.55ns and `[1, 2] ++ 500_items` takes 2.48ns, basically indistinguishable and a million times faster than our calls to the DB. So I'll edit for readability.

Reply:
I ran a small benchmark as well

```

##### With input large #####
Name              ips        average  deviation         median         99th %
concat         7.02 M      142.42 ns  ±5708.31%           0 ns        1000 ns
prepend        6.32 M      158.19 ns  ±6303.88%           0 ns        1000 ns

Comparison:
concat         7.02 M
prepend        6.32 M - 1.11x slower +15.77 ns

##### With input medium #####
Name              ips        average  deviation         median         99th %
concat         6.79 M      147.32 ns ±11783.25%           0 ns        1000 ns
prepend        6.54 M      152.97 ns ±11337.75%           0 ns        1000 ns

Comparison:
concat         6.79 M
prepend        6.54 M - 1.04x slower +5.65 ns

##### With input small #####
Name              ips        average  deviation         median         99th %
prepend        7.16 M      139.67 ns  ±1950.54%           0 ns        1000 ns
concat         6.60 M      151.42 ns  ±1906.08%           0 ns        1000 ns

Comparison:
prepend        7.16 M
concat         6.60 M - 1.08x slower +11.75 ns
```